lms/djangoapps/course_api/blocks/transformers/milestones.py

```diff
@@ -35,22 +45,69 @@ def collect(cls, block_structure):
         block_structure.request_xblock_fields('is_proctored_enabled')
         block_structure.request_xblock_fields('is_practice_exam')
         block_structure.request_xblock_fields('is_timed_exam')
+        block_structure.request_xblock_fields('entrance_exam_id')
+
+    def transform(self, usage_info, block_structure):
+        """
+        Modify block structure according to the behavior of milestones and special exams.
+        """
+
+        def add_special_exam_info(block_key):
```

Comment:
Nit: I was thinking of this as a separate top-level method so that it reduces the complexity of the ``transform`` method. It also makes it easier to move the method to a different file.